### PR TITLE
Feat/available after

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed issues on https://github.com/0chain/gosdk/pull/331

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber: https://github.com/0chain/blobber/pull/600
- system_test: https://github.com/0chain/system_test/pull/216
- zboxcli: https://github.com/0chain/zboxcli/pull/194
- gosdk: https://github.com/0chain/gosdk/pull/418

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feat/available_after  |
| zboxcli  | feat/available_after  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feat/available_after  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: core/common/time.go [hunk 1]
```diff
@@ -22,3 +25,51 @@ func (t Timestamp) Within(seconds Timestamp) bool {
 func (t Timestamp) ToTime() time.Time {
 	return time.Unix(int64(t), 0)
 }
+
+var ErrInvalidTime = errors.New("Invalid time")
```
Comment (peterlimg) [inline core/common/time.go]: Use lower case for error messages

File: wasmsdk/blobber.go [hunk 2]
```diff
@@ -272,7 +273,18 @@ func Share(allocationID, remotePath, clientID, encryptionPublicKey string, expir
...
-	ref, err := allocationObj.GetAuthTicket(remotePath, fileName, refType, clientID, encryptionPublicKey, int64(expiration))
+	availableAt := time.Now()
+
+	if len(availableAfter) > 0 {
+		if aa, err := common.ParseTime(availableAt, availableAfter); err != nil {
+			PrintError(err.Error())
+			return "", err
+		} else {
+			availableAt = *aa
+		}
```
Comment (peterlimg) [inline wasmsdk/blobber.go:283]: The `else` is unnecessary

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: wasmsdk/blobber.go (hunk 2 above)
Comment (cnlangzi) [reply wasmsdk/blobber.go:283]: fixed

File: core/common/time.go (hunk 1 above)
Comment (cnlangzi) [reply core/common/time.go]: fixed

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feat/available_after  |
| zboxcli  | feat/available_after  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feat/available_after  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**